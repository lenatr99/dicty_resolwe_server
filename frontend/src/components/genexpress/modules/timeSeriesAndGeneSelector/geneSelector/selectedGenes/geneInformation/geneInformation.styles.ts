import styled from 'styled-components';

export const GeneInformationContainer = styled.div`
    min-width: 240px;
    max-width: 300px;
    border: 1px solid;
    padding: ${({ theme }): string => theme.spacing(1)};
    background-color: ${({ theme }): string => theme.palette.background.paper};
`;

export const GeneInformationHeader = styled.div`
    display: flex;
    flex-flow: row nowrap;
    justify-content: space-between;
    margin-bottom: 10px;
`;

export const GeneInformationTitle = styled.h3`
    margin: 0;
    display: flex;
    align-self: center;
`;

export const GeneInformationLabel = styled.div`
    color: #bdbdbd;
    margin-bottom: 5px;
`;

export const GeneInformationValue = styled.div`
    margin-bottom: 10px;
`;

export const DictyBaseLogo = styled.img`
    height: 25px;
`;
