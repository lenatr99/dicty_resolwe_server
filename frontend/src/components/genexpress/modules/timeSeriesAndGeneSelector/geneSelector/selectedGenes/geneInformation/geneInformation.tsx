import React, { ReactElement } from 'react';
import Button from '@mui/material/Button';
import {
    GeneInformationContainer,
    GeneInformationTitle,
    GeneInformationValue,
    GeneInformationLabel,
    DictyBaseLogo,
    GeneInformationHeader,
} from './geneInformation.styles';
import { Gene } from 'redux/models/internal';
import dictyBaseLogo from 'images/dictybase_logo2.jpg';

export type GeneInformationProps = {
    gene: Gene;
    highlighted?: boolean;
    onHighlight?: () => void;
    onUnhighlight?: () => void;
};
export const GeneInformation = ({
    gene,
    onHighlight,
    onUnhighlight,
    highlighted,
}: GeneInformationProps): ReactElement => {
    return (
        <GeneInformationContainer>
            <GeneInformationHeader>
                <GeneInformationTitle>Gene information</GeneInformationTitle>
                {highlighted ? (
                    <Button
                        onClick={onUnhighlight}
                        color="secondary"
                        variant="contained"
                        size="small"
                    >
                        Unhighlight
                    </Button>
                ) : (
                    <Button
                        onClick={onHighlight}
                        color="secondary"
                        variant="contained"
                        size="small"
                    >
                        Highlight
                    </Button>
                )}
            </GeneInformationHeader>
            <p>{gene.name}</p>
            <div>
                <GeneInformationLabel>Gene ID:</GeneInformationLabel>
                <GeneInformationValue>{gene.feature_id}</GeneInformationValue>
            </div>
            <div>
                <GeneInformationLabel>Alternatives:</GeneInformationLabel>
                <GeneInformationValue>{gene.aliases?.join(', ')}</GeneInformationValue>
            </div>
            <div>
                <GeneInformationLabel>Database link:</GeneInformationLabel>
                <GeneInformationValue>
                    <a
                        href={`http://dictybase.org/gene/${gene.feature_id}`}
                        rel="noopener noreferrer"
                        target="_blank"
                    >
                        Open in dictyBase
                        <DictyBaseLogo src={dictyBaseLogo} alt="Open in dictyBase." />
                    </a>
                </GeneInformationValue>
            </div>
            <div>
                <GeneInformationLabel>Description:</GeneInformationLabel>
                <GeneInformationValue>{gene.description}</GeneInformationValue>
            </div>
        </GeneInformationContainer>
    );
};

export default GeneInformation;
