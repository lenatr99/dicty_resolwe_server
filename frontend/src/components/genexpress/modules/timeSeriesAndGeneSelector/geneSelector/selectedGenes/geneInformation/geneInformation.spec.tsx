import React from 'react';
import { screen, fireEvent } from '@testing-library/react';
import { vi } from 'vitest';
import { GeneInformation } from './geneInformation';
import { customRender } from 'tests/test-utils';
import { generateGene } from 'tests/mock';

const gene = generateGene(0);

describe('geneInformation', () => {
    it('should render as default snapshot', () => {
        const { asFragment } = customRender(
            <GeneInformation
                gene={gene}
                highlighted={false}
                onHighlight={(): void => {}}
                onUnhighlight={(): void => {}}
            />,
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('should show Unhighlight button if gene is highlighted', () => {
        customRender(
            <GeneInformation
                gene={gene}
                highlighted
                onHighlight={(): void => {}}
                onUnhighlight={(): void => {}}
            />,
        );

        screen.getByText('Unhighlight');
    });

    it('should show Unhighlight button if gene is highlighted', () => {
        customRender(
            <GeneInformation
                gene={gene}
                highlighted={false}
                onHighlight={(): void => {}}
                onUnhighlight={(): void => {}}
            />,
        );

        screen.getByText('Highlight');
    });

    it('should call onHighlight when user clicks Highlight', () => {
        const mockedOnHighlight = vi.fn();
        customRender(
            <GeneInformation
                gene={gene}
                highlighted={false}
                onHighlight={mockedOnHighlight}
                onUnhighlight={(): void => {}}
            />,
        );

        fireEvent.click(screen.getByText('Highlight'));

        expect(mockedOnHighlight).toBeCalled();
    });

    it('should call onUnhighlight when user clicks Unhighlight', () => {
        const mockedOnUnhighlight = vi.fn();
        customRender(
            <GeneInformation
                gene={gene}
                highlighted
                onHighlight={(): void => {}}
                onUnhighlight={mockedOnUnhighlight}
            />,
        );

        fireEvent.click(screen.getByText('Unhighlight'));

        expect(mockedOnUnhighlight).toBeCalled();
    });
});
