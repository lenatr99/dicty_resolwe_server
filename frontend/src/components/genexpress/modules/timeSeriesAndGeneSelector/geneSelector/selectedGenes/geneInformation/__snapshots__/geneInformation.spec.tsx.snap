// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`geneInformation > should render as default snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gJhJTp kMHQvz"
  >
    <div
      class="sc-elDIKY naXCM"
    >
      <h3
        class="sc-fQpRED givOwf"
      >
        Gene information
      </h3>
      <button
        class="MuiButtonBase-root-ocQuz kiSjtm MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorSecondary MuiButton-root-jMlkZG ceGgqh MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorSecondary"
        tabindex="0"
        type="button"
      >
        Highlight
        <span
          class="MuiTouchRipple-root-huKFMv dxeyKA MuiTouchRipple-root"
        />
      </button>
    </div>
    <p>
      TestGene0
    </p>
    <div>
      <div
        class="sc-dsLQwm jsjqqF"
      >
        Gene ID:
      </div>
      <div
        class="sc-iKTcqh cRJFCK"
      >
        0
      </div>
    </div>
    <div>
      <div
        class="sc-dsLQwm jsjqqF"
      >
        Alternatives:
      </div>
      <div
        class="sc-iKTcqh cRJFCK"
      >
        TestGene0_alias1, TestGene0_alias2
      </div>
    </div>
    <div>
      <div
        class="sc-dsLQwm jsjqqF"
      >
        Database link:
      </div>
      <div
        class="sc-iKTcqh cRJFCK"
      >
        <a
          href="http://dictybase.org/gene/0"
          rel="noopener noreferrer"
          target="_blank"
        >
          Open in dictyBase
          <img
            alt="Open in dictyBase."
            class="sc-gnpbhQ kdvbuV"
            src="/images/dictybase_logo2.jpg"
          />
        </a>
      </div>
    </div>
    <div>
      <div
        class="sc-dsLQwm jsjqqF"
      >
        Description:
      </div>
      <div
        class="sc-iKTcqh cRJFCK"
      >
        This is a test gene 0
      </div>
    </div>
  </div>
</DocumentFragment>
`;
